src/app/regular-giving/regular-giving.component.ts

```diff
@@ -62,6 +62,13 @@ export class RegularGivingComponent implements OnInit {
     this.donor = donor
 
     this.campaign = this.route.snapshot.data.campaign;
+
+    if ( !this.campaign.isRegularGiving ) {
+      this.router.navigate(['/my-account']);
+      throw new Error("Campaign " + this.campaign.id + " is not a regular giving campaign");
+      return;
```

Comment:
What's the reason for navigating to my-account? I think it would make sense to either display an error on the page here, or maybe navigate to the site homepage like we do on what would be 404 errors on other sites.

Reply:
Thank you @bdsl , yes I wasn't sure either my-account or the regular campaign page or homepage, picked my-account just to make sure the redirect works and that's easy to change. But you're right we should probably display error message informing the user that this is not a regular giving campaign, but encourage to make one off donation and add a link to this campaign page?

Reply:
Thanks. Let's either keep this or the code you've put in the template, we don't need both, do we? I think the template change is fine so we could delete this code.

Also I forgot to say before it doesn't make sense to have `throw` and then `return` on the last line - any statement directly after throw is dead code.

Reply:
Thank you @bdsl , I think I left the `return` by mistake trying different things to stop the non-regular campaign from rendering in ngOninit. I'll remove this and navigation, but do you think we should keep the exception to show error in the console?

Reply:
We can but I don't think we need it when the error is shown on the page. Can we take out the `this.router.navigate(['/my-account']);` anyway?